### ISSUE TITLE
No realtime updates with findQuery

### ISSUE THREAD

**Issue by ahaurw01** [firebase/emberfire#235]
The EmberFire guides make it sound like all adapter operations will result in automatic updates. I have found that anything that returns an [`AdapterPopulatedRecordArray`](http://emberjs.com/api/data/classes/DS.AdapterPopulatedRecordArray.html) will NOT update as you or other clients make changes to the underlying data. For example, this route's model will not update as the underlying cat data changes.

```javascript
export default Ember.Route.extend({

  model: function (params) {
    return this.store.find('cat', {
      orderBy: 'breed',
      equalTo: params.breed
    });
  }

}
```

To work around this bug, I have done the following to make sure I get my hands on a `RecordArray` instead of the dreaded `AdapterPopulatedRecordArray`.

```javascript
export default Ember.Route.extend({

  model: function (params) {
    var store = this.store;
    return store.find('cat', {
      orderBy: 'breed',
      equalTo: params.breed
    }).then(function () {
      return store.filter('cat', function (cat) {
        return cat.get('breed') === params.breed;
      });
    });
  }

}
```

Is this the preferred method of obtaining auto-updating arrays for queries? It'd be awesome if the guides mentioned something about this, as I see it being quite common to want to fetch a subset of records instead of the whole enchilada. (`store.find('cat')` returns a `RecordArray` that updates automatically.)

**Comment by oskarrough** [comment]
I'm seeing the same issue and have to filter by model as @ahaurw01 before it works.

**Comment by tstirrat** [comment]
Yes this is definitely something that needs to be added.

You can shorten the workaround like to use `store.filter(type, query[, func])`:

```js
export default Ember.Route.extend({
  model: function (params) {

    return store.filter('cat', {
      orderBy: 'breed',
      equalTo: params.breed
    }, function (cat) {
      return cat.get('breed') === params.breed;
    });

  }
});
```

**Comment by ahaurw01** [comment]
Nice, thanks for the tip @tstirrat. Wasn't aware of that `filter` shorthand.

**Comment by abrinkman** [comment]
I am also seeing this issue.  Everything is fine with a basic find query.

```javascript
model: function() {
  return this.store.find('message');
}
```

But I lose realtime updates if I add in a sort:

```javascript
model : function()  {
  return this.store.find('message',  {
    orderBy: 'timestamp'
  });
}
```

**Comment by bryanveloso** [comment]
Is there any workaround for @abrinkman's case?

**Comment by tstirrat** [comment]
I am working on truly realtime queries that will maintain their ordering and sync. As for a workaround, you can use a raw firebase query that populates a standard ember array when its `value` changes:

```js
var adapter = this.store.adapterFor('message');
adapter._getRef('message')
  .orderByChild('timestamp')
  .on('value', (snapshot) => {
    var requests = [];
    snapshot.forEach((childSnapshot) => {
      requests.push( this.store.find('message', childSnapshot.key()) );
    });
    Ember.RSVP.all(requests).then((tracks) => {
      this.set('orderedMessages', tracks); // might need to wrap inside Ember.run here
    });
  });
```

[this comment](https://github.com/firebase/emberfire/issues/243#issuecomment-94038081) has some more discussion

As a side note, if you wish to order things by the time the record was created, the id (firebase key/pushId) is actually [time ordered](https://www.firebase.com/blog/2015-02-11-firebase-unique-identifiers.html).

**Comment by oskarrough** [comment]
@tstirrat, tried your last example which works and updates as new records are created. But it doesn't remove them as they're destroyed. Any ideas?

**Comment by tstirrat** [comment]
Do you mean to that the array `this.get('orderedMessages')` does not remove items as they are removed on the server?

This seems improbable because each time the query updates it's `value` we are getting the entire list of entries in the array and there should be no *removed records* in this payload. So I wonder if some other race condition is happening.

**Comment by tstirrat** [comment]
Realtime updates for queries is released in version 1.4.7
